### PR TITLE
Add direction parsing elements `¨^` and `¨□`

### LINKED ISSUE #452: direction parsing
A digraph to convert strings of `>v<^` to [0,1,2,3]

### PR DESCRIPTION
**Pull request by pxeger**
Closes #452

### CODE REVIEW

**Review by Lyxal — APPROVED** [review]
Looks good to me, but why the explicit `+1`s in the vectors?

**Comment by pxeger** [comment]
Just to make it line up nicely with the `-1`s so it's easier to read, and make it, as you say, more explicit.

File: vyxal/elements.py [hunk 1]
```diff
@@ -2585,6 +2585,38 @@ def parity(lhs, ctx):
     }.get(ts, lambda: vectorise(parity, lhs, ctx=ctx))()
 
 
+def parse_direction_arrow_to_integer(lhs, ctx):
+    """Element ¨^
+    (str) -> map characters in `>^<v` to integers
+    """
+    ts = vy_type(lhs)
+    if ts is str and len(lhs) == 1:
+        return {
```
Comment (cgccuser) [inline vyxal/elements.py:2594]: Why not manually vectorise for strings here instead? I feel like the intent may be clearer that way.

File: vyxal/elements.py [hunk 2]
```diff
@@ -2585,6 +2585,38 @@ def parity(lhs, ctx):
...
+    else:
+        return vectorise(parse_direction_arrow_to_integer, lhs, ctx=ctx)()
+
+
+def parse_direction_arrow_to_vector(lhs, ctx):
+    """Element ¨^
+    (str) -> map characters in `>^<v` to direction vectors
+    """
+    ts = vy_type(lhs)
+    if ts is str and len(lhs) == 1:
```
Comment (cgccuser) [inline vyxal/elements.py:2609]: Same here

**Review by cgccuser — DISMISSED** [review]

File: vyxal/elements.py (hunk 1 above)
Comment (pxeger) [reply vyxal/elements.py:2594]: Why not manually vectorise for lists here as well? Because we already have a function for it.

Comment (cgccuser) [reply vyxal/elements.py:2594]: Strings aren't lists, though, and I don't feel comfortable relying on vectorise treating strings as iterables.

**Review by cgccuser — APPROVED** [review]
\*grumble grumble\*

**Pull request merged**